future-apps/payroll/contracts/Payroll.sol

```diff
@@ -594,21 +598,44 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
         emit TerminateEmployee(_employeeId, employee.accountAddress, _endDate);
     }
 
-    function _getOwedSalary(uint256 _employeeId, uint64 _date) internal view returns (uint256) {
+    /**
+     * @dev Calculates the date timestamp corresponding to the requested paying amount based on the employee's last payroll date.
+     * @param _employeeId Employee's identifier.
+     * @param _payedAmount Amount payed to the employee to query the timestamp of.
+     * @return The date timestamp in seconds corresponding to the given paying amount based on the employee's last payroll date.
+     */
+    function _getLastPayroll(uint256 _employeeId, uint256 _payedAmount) internal view returns (uint64) {
+        Employee storage employee = employees[_employeeId];
+        uint64 timeDiff = _payedAmount.div(employee.denominationTokenSalary).toUint64();
```

Comment:
As I said [here](https://github.com/aragon/aragon-apps/pull/744#discussion_r270079562), I think than better than this check, I would make sure that is not greater than `getTimestamp64()`. I guess that we could then remove the `Uint256Helpers` dependency.